src/components/PreferencesBodyVirtualMachine.vue

```diff
@@ -0,0 +1,67 @@
+<script lang="ts">
+import os from 'os';
+import { ipcRenderer } from 'electron';
+
+import Vue from 'vue';
+import { defaultSettings } from '@/config/settings';
+import SystemPreferences from '@/components/SystemPreferences.vue';
+
+export default Vue.extend({
+  name:       'preferences-body-virtual-machine',
+  components: { SystemPreferences },
+  data() {
+    return { settings: defaultSettings };
+  },
+  computed:   {
+    hasSystemPreferences(): boolean {
+      return !os.platform().startsWith('win');
+    },
+    availMemoryInGB(): number {
+      return Math.ceil(os.totalmem() / 2 ** 30);
+    },
+    availNumCPUs(): number {
+      return os.cpus().length;
+    },
+  },
+  beforeMount() {
+    ipcRenderer.on('settings-read', (event, settings) => {
+      this.settings = settings;
+    });
+    ipcRenderer.send('settings-read');
+  },
+  methods: {
+    handleUpdateMemory(value: number) {
+      this.settings.kubernetes.memoryInGB = value;
+      ipcRenderer.invoke('settings-write',
+        { kubernetes: { memoryInGB: value } });
+    },
+    handleUpdateCPU(value: number) {
+      this.settings.kubernetes.numberCPUs = value;
+      ipcRenderer.invoke('settings-write',
+        { kubernetes: { numberCPUs: value } });
+    },
+  }
+});
+</script>
+
+<template>
+  <div class="preferences-content">
+    <system-preferences
```

Comment:
We'll want to restyle this later: it is the only thing with borders around the controls.

Reply:
I didn't make changes here because I didn't want to affect the components as they are represented in the current Kubernetes Settings. I added #2464 so we don't let this slip.